nextcord/ext/help_commands/embedded.py

```diff
@@ -0,0 +1,196 @@
+
+from nextcord.ext.commands import Cog, Command, Group, HelpCommand
+
+from nextcord import Embed, Message
+
+
+class EmbeddedHelpCommand(HelpCommand):
+    """A help command implementation using embeds.
+    This inherits from :class:`HelpCommand`.
+    It extends it with the following attributes.
+
+    Attributes
+    ------------
+    dm_help: Optional[:class:`bool`]
+        A bool that indicates if the output should only be sent via DM instead of the channel.
+        Defaults to False.
+    default_color: :Optional[class:`int`]
+        The default color of all embeds.
+        Defaults to 0xFFFFFF (White).
+    main_embed_title: Optional[:class:`str`]
+        The title for the embed which gets sent by invoking `help`.
+        Defaults to "Overview about Cogs, Groups and Commands".
+    main_embed_description: Optional[:class:`str`]
+        The description for the embed which gets sent by invoking `help`.
+        Defaults to an empty string.
+    main_embed_color: Optional[:class:`int`]
+        The color for the embed which gets sent by invoking `help`.
+        Defaults to default_color.
+    command_embed_title: Optional[:class:`str`]
+        The title for the embed which gets sent by `help <command>`.
+        Defaults to "Command help".
+    command_embed_description: Optional[:class:`str`]
+        The description for the embed which gets sent by invoking `help <command>`.
+        Defaults to an empty string.
+    command_embed_color: Optional[:class:`int`]
+        The color for the embed which gets sent by invoking `help <command>`.
+        Defaults to default_color.
+    group_embed_title: Optional[:class:`str`]
+        The title for the embed which gets sent by invoking `help <group>`.
+        Defaults to "Group help".
+    group_embed_description: Optional[:class:`str`]
+        The description for the embed which gets sent by invoking `help <group>`.
+        Defaults to "If a parameter is surrounded by `<>`, it is a `required` parameter\nIf a parameter is surrounded by `[]`, it is an `optional` parameter.".
+    group_embed_color: Optional[:class:`int`]
+        The color for the embed which gets sent by invoking `help <group>`.
+        Defaults to default_color.
+    cog_embed_title: Optional[:class:`str`]
+        The title for the embed which gets sent by invoking `help <cog>`.
+        Defaults to "Cog help".
+    cog_embed_description: Optional[:class:`str`]
+        The description for the embed which gets sent by invoking `help <cog>`.
+        Defaults to an empty string.
+    cog_embed_color: Optional[:class:`int`]
+        The color for the embed which gets sent by invoking `help <cog>`.
+        Defaults to default_color.
+    """
+
+    def __init__(
+        self,
+        *,
+        dm_help: bool = False,
+        default_color: int = 0xFFFFFF,
+        main_embed_title: str = "Overview about Cogs, Groups and Commands",
+        main_embed_description: str = "",
+        main_embed_color: int = None,
```

Comment:
Any parameter or attribute that can be `None` must be type annotated with `typing.Optional`
```suggestion
        main_embed_color: Optional[int] = None,
```